### PR TITLE
Update SpatialEpigenome.rst

### PR DESCRIPTION
**Pull request by tamarajafar**
updates.

### CODE REVIEW

File: docs/SpatialEpigenome.rst [hunk 1]
```diff
@@ -279,19 +269,14 @@ in specific regions of the tissue, visualize it's spatial distribution, and gain
 ------------------------------------------------------------------------
 
 **Add group coverages**
-Use the addGroupCoverages function to call on the input ArchR project object, proj_in_tissue, to add group coverages to the object. The groupBy 
-parameter specifies which metadata column to group the coverages by, in this case 'Clusters'.  
-This function calculates the average coverage of each peak in a single-cell RNA-seq dataset, grouped by 'Clusters'. The resulting object contains the 
-average coverage of each peak in each group, along with metadata about the peaks and the groups. This is done to obtain an understanding of the 
-coverage of peaks across different clusters to provide insights into various cluster characteristics.::
+
+It's important to understand how gene expression varies across different cell types or clusters. The addGroupCoverages function calculates the average coverage of each genomic region (or "peak") in the dataset, grouped by a specific metadata column (in this case, Clusters). By doing so, it allows us to compare the average expression of each peak across different cell types, providing insights into various cluster characteristics.::
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: This function is more so for creating pseudo-bulk replicates to account for binary nature of single cell ATAC data and to prepare our data for peak-calling. It also helps with determining how reproducible peak calling is in the next step.In ATAC data, we dont have gene expression data however we do have gene score information which attempts to assigns an accessibility "score" to each gene according to the number of peaks near its TSS

File: docs/SpatialEpigenome.rst [hunk 2]
```diff
@@ -279,19 +269,14 @@ in specific regions of the tissue, visualize it's spatial distribution, and gain
...
-proj_in_tissue to add reproducible peak sets to the object. The groupBy, pathToMacs2, and genomeSize parameters are used to specify the metadata column 
-to group the peaks by, the path to the macs2 program, and the size of the genome. The force parameter is set to TRUE to force re-running the peak 
-calling even if it has already been performed. This step is done to identify reproducible peaks across different clusters.::
+
+It's also important to identify genomic regions that are consistently expressed across multiple samples. These are called "peaks" and can be identified using software like MACS2. The addReproduciblePeakSet function to call peaks on our dataset (proj_in_tissue) and add the results to the object. To use this function, we pass in the ArchRProj object, the name of the metadata column to group the peaks by (Clusters), the path to the MACS2 program, and the size of the genome we're analyzing. We also set the force parameter to TRUE, which tells the function to re-run peak calling even if it's already been done (useful if we've made changes to our dataset).::
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: Better to specify **accessible** genomic regions. Also, change "The addReproduciblepeakSet function to ..." to  "We call the addReproduciblePeakSet function to ..."

File: docs/SpatialEpigenome.rst [hunk 3]
```diff
@@ -304,18 +289,16 @@ calling even if it has already been performed. This step is done to identify rep
...
 
    proj_in_tissue <- addPeakMatrix(proj_in_tissue)
 
 
 Motif enrichment (Deviation)
 ----------------------------------------
 **Add motif annotations** 
-First, we check if motif annotations are already present in the project. If not, we add motif annotations to the ArchR project using the addMotifAnnotations function. If the data species is "hg38" or "mm10", we use the "cisbp" motif set. Otherwise, we use the "encode" motif set and obtain the species information from the project's genome.::
+
+Check if motif annotations are already present in the project. If not, use the addMotifAnnotations function to add them to the ArchR project. The motif set used will depend on the data species. If the species is "hg38" or "mm10", the "cisbp" motif set is used. Otherwise, the "encode" motif set is used and the species information is obtained from the project's genome.::
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: You can get rid of the if-else code block and just keep the line of code to add motif annotations

File: docs/SpatialEpigenome.rst [hunk 4]
```diff
@@ -326,13 +309,12 @@ First, we check if motif annotations are already present in the project. If not,
...
 **Add background peaks** 
-Use the addBgdPeaks() function to add background peak information. This function takes the ArchRProj object as an input, along with the force argument, which is set to TRUE so that it'll overwrite any existing background peak information in the object. ::
+Use the addBgdPeaks() function to add background peak information to the ArchRProj. The force argument is set to TRUE so that it'll overwrite any existing background peak information in the object. ::
 
    proj_in_tissue <- addBgdPeaks(proj_in_tissue, force = TRUE)
 
 **Add deviations matrix**
-Add a matrix of deviations using the addDeviationsMatrix() function. This function takes the ArchRProj object as an input, along with the 
-peakAnnotation argument, which specifies the name of the peak annotations to use when calculating the deviations. ::
+Use the addDeviationsMatrix() function to add a matrix of deviations to the ArchRProj object. The peakAnnotation argument specifies the name of the peak annotations to use when calculating the deviations.::
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: Change "add a matrix of deviations" to "add a matrix of per-cell deviations "

File: docs/SpatialEpigenome.rst [hunk 5]
```diff
@@ -386,12 +367,17 @@ and assign the resulting list of motifs to the variable motifs. We do this to cr
...
+   source(paste0(script.dir, "/", "getGeneScore_ArchR.R"))
+   source(paste0(script.dir, "/", "SpatialPlot_new.R"))
+   source(paste0(script.dir, "/", "SpatialDimPlot_new.R"))
 
 Deviation scores and matrices
 ----------------------------------------
-
-Apply getDeviation_ArchR to the modified Seurat object and the list of motifs, along with the result of applying the getImputeWeights function to the 
-modified Seurat object. Assign the result to the variable dev_score. ::
+Compute deviation scores for the motifs of interest using the getDeviation_ArchR() function. The function takes the ArchR project object, the list of motifs, and the imputed weights computed with getImputeWeights(). Assign the resulting deviation scores to the variable dev_scores. ::
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: This functions gets the deviation  scorematrix from the archR project. Deviation scores are computed before.

File: docs/SpatialEpigenome.rst [hunk 6]
```diff
@@ -450,19 +436,29 @@ Save the combined plot as a PNG image. ::
...
- mat <- TFBSTools::as.matrix(motif_ID)
- probmat <- exp(mat) * matrix(TFBSTools::bg(motif_ID), nrow = nrow(mat), ncol = ncol(mat),  byrow = FALSE)
- logo_list[[i]] <- ggseqlogo(probmat)
-  }
-  logo_plots <- wrap_plots(logo_list, ncol = 3)
-
-      png(file="logos.png", width = 8, height=ceiling(length(motifs)/3)*1.5, unit="in", res = 300)
+   
+ **Generate List of ggplot Objects for Each Motif Plot**
+Create individual motif plots by by generating a list of ggplot objects for each motif plot. It iterates over the row names of spatial.obj, extracts the motif PWM for each row, and converts it to a probability matrix. It then uses the probability matrix to create a ggplot object for the motif plot. The reason for this step is to create individual motif plots that can be combined into a single plot.::
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: You can get rid of this code block/section which. computes the seqlogos plots (these are not motif plots)

File: docs/SpatialEpigenome.rst (hunk 1 above)
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: This function is more so for creating pseudo-bulk replicates to account for binary nature of single cell ATAC data and to prepare our data for peak-calling. It also helps with determining how reproducible peak calling is in the next step.In ATAC data, we dont have gene expression data however we do have gene score information which attempts to assigns an accessibility "score" to each gene according to the number of peaks near its TSS

**Review by poorvi4 — CHANGES_REQUESTED** [review]
Please make the requested changes as given in the review comments.

**Comment by tamarajafar** [comment]
final review needed to merge! added requests

File: docs/SpatialEpigenome.rst [hunk 7]
```diff
@@ -450,19 +436,29 @@ Save the combined plot as a PNG image. ::
...
+
```
Comment (poorvi4) [inline docs/SpatialEpigenome.rst]: Please delete the ggseqlogo section. This section is used to create the seqlogo plots which are different from the motif plots. Also the image to show the respective marker motifs are missing from this commit however were found in previous merges.

**Review by poorvi4 — CHANGES_REQUESTED** [review]
See comments.

**Review by poorvi4 — APPROVED** [review]

**Pull request merged**